packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -335,11 +406,10 @@ class _DismissibleState extends State<Dismissible> with TickerProviderStateMixin
     super.dispose();
   }
 
-  bool get _directionIsXAxis {
-    return widget.direction == DismissDirection.horizontal
-        || widget.direction == DismissDirection.endToStart
-        || widget.direction == DismissDirection.startToEnd;
-  }
+  bool get _directionIsXAxis => switch (widget.direction) {
+    DismissDirection.horizontal || DismissDirection.endToStart || DismissDirection.startToEnd => true,
+    DismissDirection.vertical || DismissDirection.up || DismissDirection.down || DismissDirection.none => false
```

Comment:
Thanks for converting from a `==` chain to an exhaustive switch expression!

<br>

Might as well add a [trailing comma](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-a-trailing-comma-for-arguments-parameters-and-list-items-but-only-if-they-each-have-their-own-line) here:

```suggestion
    DismissDirection.vertical || DismissDirection.up || DismissDirection.down || DismissDirection.none => false,
```